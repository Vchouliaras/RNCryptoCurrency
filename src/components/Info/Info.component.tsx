import * as React from 'react'
import { View, Text } from '@shoutem/ui'

import Styles from './Info.styles'
import CryptoContext from '../../context'
import { Coins } from '../../utils'

import { IInfoState } from '../../types'

export default class Info extends React.Component<{}, IInfoState> {
  static contextType = CryptoContext

  state = { seconds: 0 }

  private timerID: number = 0

  tick = () => {
    this.setState(state => ({ seconds: state.seconds + 1 }))
  }

  componentDidUpdate(_prevProps: any, prevState: IInfoState) {
    if (prevState.seconds === this.state.seconds && this.state.seconds > 0) {
      this.setState({ seconds: 0 })
    }
  }

  componentDidMount() {
    this.timerID = setInterval(() => this.tick(), 1000)
  }

  componentWillUnmount() {
    clearInterval(this.timerID)
  }

  render() {
    const areProductsEmpty = Coins.areProductsEmpty(this.context)

    if (areProductsEmpty) {
      return null
    }

    return (
      <View style={Styles.container}>
        <Text styleName="h-center">{`Last update ${this.state.seconds}s ago`}</Text>
      </View>
    )
  }
}
