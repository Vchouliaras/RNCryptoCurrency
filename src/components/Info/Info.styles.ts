import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    marginTop: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
})

export default styles
