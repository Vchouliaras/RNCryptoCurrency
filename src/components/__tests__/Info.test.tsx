import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Info from '../Info/Info.component'

describe('Info Component', () => {
  beforeEach(() => jest.useFakeTimers())

  it('renders correctly', () => {
    const wrapper: ShallowWrapper = shallow(<Info />)

    expect(wrapper).toMatchSnapshot()
  })

  it('componentDidMount: starts the timer', () => {
    const wrapper: ShallowWrapper = shallow(<Info />)

    jest.runOnlyPendingTimers()

    expect(wrapper.state('seconds')).toBeGreaterThan(0)
  })
})
