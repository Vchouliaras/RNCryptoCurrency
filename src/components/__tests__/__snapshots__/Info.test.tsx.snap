// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Component renders correctly 1`] = `""`;
